mos-platform/common/include/fixed_point.h

```diff
@@ -0,0 +1,301 @@
+  /// Update just the integral portion
+  [[clang::always_inline]] constexpr void SetI(IntType v) { 
+    val = (val & ~INT_MASK) | static_cast<StorageType>((v) & (INT_MASK >> FSize)) << FSize;
+  }
+  /// Update just the fractional portion
+  [[clang::always_inline]] constexpr void SetF(FracType v) { 
+    val = (val & ~FRAC_MASK) | (v & FRAC_MASK);
+  }
+  /// Update the entire value
+  [[clang::always_inline]] constexpr void Set(StorageType v) { val = v; }
+
+  // Operator overloads
+  [[clang::always_inline]] constexpr Value& operator=(Value o) {
+    val = o.val;
+    return *this;
+  }
+
+  [[clang::always_inline]] constexpr Value operator +(const Value& o) const {
+    Value<ISize, FSize> n = val;
+    n += o;
+    return n;
+  }
+  [[clang::always_inline]] constexpr Value& operator +=(const Value& o) {
+    val += o.val;
+    return *this;
+  }
+
+  [[clang::always_inline]] constexpr Value operator -(const Value& o) const {
+    Value<ISize, FSize> n = val;
+    n -= o;
+    return n;
+  }
+  [[clang::always_inline]] constexpr Value& operator -=(const Value& o) {
+    val -= o.val;
+    return *this;
+  }
+
+  template<typename T, typename = typename std::enable_if<std::is_arithmetic<T>::value, T>::type>
+  [[clang::always_inline]] constexpr Value operator/(T i) {
+    Value<ISize, FSize> n{val};
+    n /= i;
+    return n;
+  }
+  template<typename T, typename = typename std::enable_if<std::is_arithmetic<T>::value, T>::type>
+  [[clang::always_inline]] constexpr Value& operator/=(T i) {
+    val /= i;
+    return *this;
+  }
+
+  template<typename T, typename = typename std::enable_if<std::is_arithmetic<T>::value, T>::type>
+  [[clang::always_inline]] constexpr Value operator*(T i) {
+    Value<ISize, FSize> n{val};
+    n *= i;
+    return n;
+  }
+  template<typename T, typename = typename std::enable_if<std::is_arithmetic<T>::value, T>::type>
+  [[clang::always_inline]] constexpr Value& operator*=(T i) {
+    val *= i;
+    return *this;
+  }
+
+  template<typename T, typename = typename std::enable_if<std::is_arithmetic<T>::value, T>::type>
+  [[clang::always_inline]] constexpr Value operator >>(T v) const {
+    Value<ISize, FSize> n = *this;
+    n >>= v;
+    return n;
+  }
+  template<typename T, typename = typename std::enable_if<std::is_arithmetic<T>::value, T>::type>
+  [[clang::always_inline]] constexpr Value& operator >>=(T v) {
+    val >>= v;
+    return *this;
+  }
+  
+  template<typename T, typename = typename std::enable_if<std::is_arithmetic<T>::value, T>::type>
+  [[clang::always_inline]] constexpr Value operator <<(T v) const {
+    Value<ISize, FSize> n = *this;
+    n <<= v;
+    return n;
+  }
+  template<typename T, typename = typename std::enable_if<std::is_arithmetic<T>::value, T>::type>
+  [[clang::always_inline]] constexpr Value& operator <<=(T v) {
+    val <<= v;
+    return *this;
+  }
+
+  [[clang::always_inline]] Value &operator++() {
```

Comment:
Hmm, I'd think that this operation would increment the integer part, not the base representation. It should be logically equivalent to `+= 1`, and right now it isn't.

It might be okay to have a break here if there's a compelling reason, but that reason should definitely be written down somewhere at the very least.